src/box/lua/connpool.lua

```diff
@@ -125,6 +129,51 @@ local function connect_to_candidates(candidates)
     return connected_candidates
 end
 
+local acquire_any_cond = nil
+
+-- The method starts connecting to the specified set of
+-- instances and returns the first one available matching
+-- the specified dynamic requirements.
+--
+-- Note: if the specified mode is 'alive' the connection
+-- returned may not have the `mode()` method available.
+local function acquire_any_candidate(instance_names, opts)
+    assert(type(opts) == 'table' and opts.mode)
+    assert(opts.mode == 'ro' or opts.mode == 'rw' or opts.mode =='alive')
+    local time_connect_end = clock.monotonic() + WATCHER_TIMEOUT
+
+    if not acquire_any_cond then
+        acquire_any_cond = fiber.cond()
+    end
+
+    for _, instance_name in pairs(instance_names) do
+        local ok, conn = pcall(connect, instance_name, {
+            wait_connected = false,
+            connect_timeout = WATCHER_TIMEOUT
+        })
+
+        if ok and not is_candidate_checked(instance_name) then
+            local function on_connect_notify()
+                acquire_any_cond:broadcast()
+            end
+
+            conn:on_connect(on_connect_notify)
+        end
+    end
+
+    while clock.monotonic() < time_connect_end do
+        for _, instance_name in ipairs(instance_names) do
+            local conn = connections[instance_name]
+            if is_connection_valid(conn, {}) and
```

Comment:
Just curious: if a connection in in a terminal state (for example, `error`), it means that `is_connection_valid` == `false`, but there is nothing to wait. Should we also check `is_connection_checked`?

Reply:
Hmm, you're right. I've added implemented the behaviour. Thank you.